### PR TITLE
Added application, system, & otel requirement pages

### PR DESCRIPTION
**Pull request by cartersocha**
Translated the requirements into markdown files. Planning on linking to these from the overall readme after the initial drafts are approved.

Requirements doc can be found here: https://docs.google.com/document/d/1ncCKFzgUf8XX7dw6a0U31RNYnPHhHg1s5cF-qYLlmEU/edit#

### CODE REVIEW

File: Requirements/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+We're building the demo application with several different target personas in mind:
+
+1. **Enthusiasts** at a company that can use the demo app as an individual to
+   advocate for OTel within their organization.
+
+2. **Developers** with specific language skills who want to see a larger picture
+   view.
+
+3. **APM Vendors** who can evaluate OTel in general or need to produce a demo of
+   their OTel capabilities for customers
```
Comment (reyang) [inline Requirements/README.md:24]:
```suggestion
   their OTel capabilities for customers.
```

File: Requirements/System-Requirements.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# System Requirements
+
+To ensure the demo runs correctly please ensure your environment meets the
+following system requirements:
+
+1. Your system must meet the local [Docker](#docker-requirements) requirements
+   or you should use your preferred Cloud Service.
+
+2. The demo must be able to work on the following Operating Systems (OS):
+   Windows, Linux, and Mac with documentation provided for each OS.
```
Comment (reyang) [inline Requirements/System-Requirements.md:10]:
```suggestion
   Linux, macOS and Windows with documentation provided for each OS.
```

File: Requirements/System-Requirements.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+## Docker Requirements
+
+Docker requirements for each Operating System can be found below:
```
Comment (reyang) [inline Requirements/System-Requirements.md:14]:
```suggestion
Docker requirements for each operating system can be found below:
```

**Review by reyang — APPROVED** [review]
LGTM.

File: Requirements/Application-Requirements.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+      above requirement.
+
+2. Application processes must be language independent.
+
+    * gRPC is preferred where available and HTTP is to be used where it is not.
+
+3. Services should be architected to be modular components that can be switched out.
+
+    * Individual services can and should be encouraged to have multiple language
+      options available.
```
Comment (mic-max) [inline Requirements/Application-Requirements.md:20]: This is more of a stretch goal right? First and foremost we should focus on achieving step 1 in this document.

File: Requirements/System-Requirements.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
+# System Requirements
+
+To ensure the demo runs correctly please ensure your environment meets the
+following system requirements:
+
+1. Your system must meet the local [Docker](#docker-requirements) requirements
```
Comment (mic-max) [inline Requirements/System-Requirements.md:6]: IMO a link to `https://docs.docker.com/desktop` here would suffice, and not need to include links to each platform
```suggestion
1. Your system must meet [Docker Desktop](https://docs.docker.com/desktop) system requirements
```

**Review by mic-max — APPROVED** [review]
LGTM

File: Requirements/Application-Requirements.md (hunk 4 above)
Comment (cartersocha) [reply Requirements/Application-Requirements.md:20]: Yes more of a stretch goal. I think we'll inherently be a bit limited around the gRPC & HTTP usage but a more mature future version should have more language options per service within reason. But it's not a hard requirement to have at least 2 per service.

**Pull request merged**